src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CollectionModelBinder.cs

```diff
@@ -133,13 +165,49 @@ public class CollectionModelBinder<TElement> : IModelBinder
             return boundCollection;
         }
 
-        // Extensibility point that allows the bound collection to be manipulated or transformed before
-        // being returned from the binder.
+        /// <summary>
+        /// Gets an <see cref="object"/> suitable for the associated property.
```

Comment:
will reword to make it clear new value should be assignable to the current `ModelExplorer.ModelType`